viv_utils/emulator_drivers.py

```diff
@@ -31,67 +30,39 @@ def __str__(self):
         return "InstructionRangeExceededError(ended at instruction 0x%08X)" % self.pc
 
 
-class Hook(LoggingObject):
-    def __init__(self):
-        super(Hook, self).__init__()
-
+class Hook:
     def hook(self, callname, emu, callconv, api, argv):
         # must return something other than None if handled
         # raise UnsupportedFunction to pass
         raise UnsupportedFunction()
 
 
-class Monitor(vivisect.impemu.monitor.EmulationMonitor, LoggingObject):
-    def __init__(self, vw):
-        vivisect.impemu.monitor.EmulationMonitor.__init__(self)
-        LoggingObject.__init__(self)
-        self._vw = vw
-        self._logger = logging.getLogger("Monitor")
-
-    def getStackValue(self, emu, offset):
-        return emu.readMemoryFormat(emu.getStackCounter() + offset, "<P")[0]
-
-    def dumpStack(self, emu, num):
-        self._logger.debug("stack: ESP: %s", hex(emu.getStackCounter()))
-        for i in range(num):
-            self.d("stack: ESP + %s: %s",
-                               hex(emu.imem_psize * i),
-                               hex(self.getStackValue(emu, emu.imem_psize * i)))
```

Comment:
old code: nobody used this

Reply:
yep

Reply:
wait, I think we do in FLOSS

Reply:
yes, but can remove it there or move it, if needed